File: app/src/main/java/com/microsoft/msftbreweryworkshop/api/BreweryApi.java
package com.microsoft.msftbreweryworkshop.api;

import com.microsoft.msftbreweryworkshop.api.model.BreweryDetail;
import com.microsoft.msftbreweryworkshop.api.model.BreweryItem;
import com.microsoft.msftbreweryworkshop.api.model.DetailResponse;
import com.microsoft.msftbreweryworkshop.api.model.ListResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface BreweryApi {
    @GET("breweries")
    public Call<ListResponse<BreweryItem>> getBreweries();

    @GET("brewery/{id}")
    public Call<DetailResponse<BreweryDetail>> getBrewery(
        @Path("id") String breweryId
    );
}

File: app/src/main/java/com/microsoft/msftbreweryworkshop/api/BreweryApi.kt
package com.microsoft.msftbreweryworkshop.api

import com.microsoft.msftbreweryworkshop.api.model.BreweryDetail
import com.microsoft.msftbreweryworkshop.api.model.BreweryItem
import com.microsoft.msftbreweryworkshop.api.model.DetailResponse
import com.microsoft.msftbreweryworkshop.api.model.ListResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface BreweryApi {
    @GET("breweries")
    fun getBreweries(): Call<ListResponse<BreweryItem>>

    @GET("brewery/{id}")
    fun getBrewery(
        @Path("id") breweryId: String
    ): Call<DetailResponse<BreweryDetail>>
}